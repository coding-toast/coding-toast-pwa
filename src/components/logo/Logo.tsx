import * as React from 'react';
import { Image } from 'react-bootstrap';

interface ILogoProps {
  fillWidth?: boolean;
}

const Logo: React.FC<ILogoProps> = (props) => {
  const { fillWidth } = props;
  const logoWidth = fillWidth ? '100%' : '12em';
  return <Image className='logoImage' style={{ width: logoWidth }} src='/logo.svg' alt='coding toast' />;
};

export default Logo;
