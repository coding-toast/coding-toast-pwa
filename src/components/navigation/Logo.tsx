import * as React from 'react'
import LogoImage from '../../images/logo.png'
import LogoStyles from './Logo.module.scss'

const Logo: React.FC = () => (
  <a href="/">
    <div className={LogoStyles.logoContainer}>
      <img className={LogoStyles.logoImage} src={LogoImage} alt="Logo" />
    </div>
  </a>
)

export default Logo
