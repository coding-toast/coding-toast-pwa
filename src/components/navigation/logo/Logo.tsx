import * as React from 'react'
import { Image } from 'react-bootstrap'
import LogoImage from '../../../images/logo.svg'
import LogoStyles from './Logo.module.scss'

interface LogoProps {
  fillWidth?: boolean
}

const Logo: React.FC<LogoProps> = (props) => {
  const { fillWidth } = props
  const logoWidth = fillWidth ? '100%' : '12em'
  return <Image className={LogoStyles.logoImage} style={{ width: logoWidth }} src={LogoImage} alt="coding toast" />
}

export default Logo
