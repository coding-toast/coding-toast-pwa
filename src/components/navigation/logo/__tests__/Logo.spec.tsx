import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Logo from '../Logo'

describe('<Logo />', () => {
  it('renders correctly without fill width', () => {
    const component = shallow(<Logo />)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('renders correctly with fill width', () => {
    const component = shallow(<Logo fillWidth />)
    expect(toJson(component)).toMatchSnapshot()
  })
})
