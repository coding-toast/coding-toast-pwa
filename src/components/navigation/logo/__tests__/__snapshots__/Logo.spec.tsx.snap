// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> renders correctly with fill width 1`] = `
<Image
  alt="coding toast"
  className="logoImage"
  fluid={false}
  rounded={false}
  roundedCircle={false}
  src="test-file-stub"
  style={
    Object {
      "width": "100%",
    }
  }
  thumbnail={false}
/>
`;

exports[`<Logo /> renders correctly without fill width 1`] = `
<Image
  alt="coding toast"
  className="logoImage"
  fluid={false}
  rounded={false}
  roundedCircle={false}
  src="test-file-stub"
  style={
    Object {
      "width": "12em",
    }
  }
  thumbnail={false}
/>
`;
